### PR TITLE
orchagent: Fixing intfsorch bug wrong reference count and remove intfs case

### PR DESCRIPTION
**Pull request by stcheng**
* IntfsEntry shall only be created the first time a new IP is assigned to it
* Remove router interface only when no IP addresses are associated with it
* Add INFO level logs to increase/decrease ref count functions
* Add more information to misc logs

### CODE REVIEW

File: orchagent/intfsorch.cpp [hunk 1]
```diff
@@ -81,20 +89,26 @@ void IntfsOrch::doTask(Consumer &consumer)
             if (m_syncdIntfses.find(alias) == m_syncdIntfses.end() ||
                 !m_syncdIntfses[alias].ip_addresses.contains(ip_prefix.getIp()))
             {
-                if (addRouterIntfs(port))
+                if (m_syncdIntfses.find(alias) == m_syncdIntfses.end())
```
Comment (stcheng) [inline orchagent/intfsorch.cpp]: fixed

File: orchagent/intfsorch.cpp [hunk 2]
```diff
@@ -33,12 +33,20 @@ sai_object_id_t IntfsOrch::getRouterIntfsId(string alias)
 
 void IntfsOrch::increaseRouterIntfsRefCount(const string alias)
 {
+    SWSS_LOG_ENTER();
+
     m_syncdIntfses[alias].ref_count++;
+    SWSS_LOG_INFO("Router interface %s ref count is increased to %d",
```
Comment (lguohan) [inline orchagent/intfsorch.cpp]: DEBUG LEVEL LOG

File: orchagent/intfsorch.cpp [hunk 3]
```diff
@@ -33,12 +33,20 @@ sai_object_id_t IntfsOrch::getRouterIntfsId(string alias)
...
+                  alias.c_str(), m_syncdIntfses[alias].ref_count);
 }
 
 void IntfsOrch::decreaseRouterIntfsRefCount(const string alias)
 {
+    SWSS_LOG_ENTER();
+
     m_syncdIntfses[alias].ref_count--;
+    SWSS_LOG_INFO("Router interface %s ref count is decreased to %d",
```
Comment (lguohan) [inline orchagent/intfsorch.cpp]: DEBUG LEVEL LOG

Comment (stcheng) [inline orchagent/intfsorch.cpp]: fixed

File: orchagent/intfsorch.cpp (hunk 2 above)
Comment (stcheng) [inline orchagent/intfsorch.cpp]: fixed

**Pull request merged**